src/components/Post.jsx

```diff
@@ -285,6 +297,17 @@ function Post(prop) {
             <p key={userComment.id}>
               <strong>{userComment.content.username}</strong>{" "}
               {userComment.content.text}
+              <span
+                onClick={(event) =>
+                  deleteComment(event, userComment)
+                }
+              >
+                {userComment.content.username === user.displayName ? (
```

Comment:
there's a catch in line 305, currently people with same usernames can delete each other's comment or someone without a username (null) can delete any comment without a username. But I think if we fix the null username issue and have unique username feature, then this would get automatically resolved. What's your call on this? Or can we check some other condition like email Id which is unique?
Also, as a safe tool use at least "user && === userComment.content.username" as the page will crash if no one is signed in

Reply:
okay 
should we do it with mail?

Reply:
okay 
I think unique username feature would do it